Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -161,12 +194,22 @@ - (void)handleRemoteNotificationsRegistered:(NSNotification *)notification
   callback(@[@(RCTSharedApplication().applicationIconBadgeNumber)]);
 }
 
-RCT_EXPORT_METHOD(requestPermissions:(NSDictionary *)permissions)
+RCT_EXPORT_METHOD(requestPermissions:(NSDictionary *)permissions
+                 resolver:(RCTPromiseResolveBlock)resolve
+                 rejecter:(RCTPromiseRejectBlock)reject)
 {
   if (RCTRunningInAppExtension()) {
+    reject(RCTErrorUnableToRequestPermissions, nil, RCTErrorWithMessage(@"Requesting push notifications is currently unavailable in an app extension"));
     return;
   }
 
+  if (self.requestPermissionsResolveBlock != nil) {
+    NSString *message = @"Cannot call requestPermissions twice before the first has returned.";
+    [NSException raise:RCTFatalExceptionName format:@"%@", message];
```

Comment:
This doesn't seem fatal. Use RCTLogError here. Use RCTFatal if you actually want to make it fatal.